File: Exercicios_guanaraba/Com_Dart/bin/ex020_Sorteando_uma_ordem_na_lista.dart
/* 
O mesmo professor do desafio anterior quer sortear a ordem de apresentação de trabalhos 
dos alunos.  
Faça um programa que leia o nome dos quatros alunos e mostre a ordem sorteada.
*/
import 'dart:io';
import 'dart:math';

void main (){

var nomes = [];

print ('Insira os nomes');
for (var i=0; i <=4; i++){

  nomes.add(stdin.readLineSync());

}

print ('Os nomes são: ' + nomes.toString());
nomes.sort();
print ('Organizando com sort: ' + nomes.toString());
nomes.shuffle();
print ('Embaralhando usando shuffle: ' + nomes.toString());

/*
Para usar as funções sort e shuffle o dart altera a variavél lista nomes
não sei se é possivél inserir junto do print, mas desta forma funciona.

Para atribuir uma lista em outra segue o exemplo abaixo.

List nomes2 = nomes;

A lista nomes2 ficou com a ultima organização, no exemplo acima foi feito pelo shuffle.
*/




}

File: Exercicios_guanaraba/Com_Ruby/exercicios/ex020_Sorteando_uma_ordem_na_Lista.rb
=begin
O mesmo professor do desafio anterior quer sortear a ordem de apresentação de trabalhos dos alunos. 
Faça um programa que leia o nome dos quatro alunos e mostre a ordem sorteada.
=end

nomes = []

for nome in (0...4)
    nomes[nome] = gets.chomp
end

puts "Organizado em ordem alfabética #{nomes.sort}"
puts "Sorteado aleatóriamente #{nomes.shuffle}"